docs/Guides/Recommendations.md

```diff
@@ -298,3 +299,17 @@ readinessProbe:
     timeoutSeconds: 3
     successThreshold: 1
     failureThreshold: 5
+```
+
+## Running Multiple Instances
+<a id="multiple"></a>
+
+There are several use-cases where running multiple Fastify
```

Comment:
It is not clear to me why this setup is better?

The `metrics` data is calculated from the `API server` itself.
If you are exposing `healthcheck` and `metrics` in a dedicated instance. Does it means that you are create an extra proxy?

```
Reverse Proxy ---> Metrics Server ---> API Server
```

Then, why don't you setup the metrics handler inside `Reverse Proxy`?

Reply:
At least our situation is that while it is possible to use an elaborate nginx setup to only expose a subset of endpoints, our platform team is asking us to expose metrics on a separate port instead, as that would reduce complexity on their end, and this seems to be a fairly popular practice in the industry.
Would you recommend otherwise?

It would be something close to this: https://github.com/SkeLLLa/fastify-metrics/issues/43#issuecomment-1174497920

Reply:
For me, two port for a single application certainly increase the complexity (including both server and applications).

Imagine when you try to scale up horizontally, which means the port open is always duplicated and shouldn't be the same.

Moreover, taking nginx as configuration example. It requires to use two different upstream instead of sharing one. It actually duplicating works.

It is completely fine if this is hard requirement for your company. But I wouldn't recommend to others.

I can give out a more solid example for why it only increase complexity in later time.

Reply:
It's not a hard requirement, it's an open discussion between engineers and platform. If one is preferable over the other, we should clearly document it and then follow it.
@mcollina Can you chime in on this?

Reply:
> our platform team is asking us to expose metrics on a separate port instead, as that would reduce complexity on their end, and this seems to be a fairly popular practice in the industry.

Is it? This is the first time I have heard of doing such a thing. In fact, it makes no sense to me. The healthcheck/metrics endpoint, on a different port, is going to give you insight into an instance _that is not processing requests_.

Reply:
Exposing a separate port is the recommended approach by prometheus.

Reply:
> Exposing a separate port is the recommended approach by prometheus.

I search a bit around and couldn't found any related article about this approach. And it is not on `prometheus` document either.
May I know where you read this information?